tests/testthat/test-file-functions.R

```diff
@@ -0,0 +1,166 @@
+library(ncdf4)
+library(PCICt)
+library(proj4)
+
+test_that("time series can be retrieved from file", {
+	f <- list(t=nc_open("test1.nc"), not=nc_open("test1.nc", readunlim=FALSE))
+	correct.data.ts.test1 <- structure(c(599227200, 599313600, 599400000, 599486400, 599572800),
+			.Dim = 5L, cal = "365", months = c(31, 28, 31, 30, 31, 30,
+					31, 31, 30, 31, 30, 31),
+			class = "PCICt", dpy = 365, tzone = "GMT", units = "secs")
+	correct.data.ts.test2 <- structure(c(599227200, 599313600, 599400000, 599486400, 599572800),
+			.Dim = 5L, cal = "365", months = c(31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31),
+			class = "PCICt", dpy = 365, tzone = "GMT", units = "secs",
+			bounds = structure(c(599184000, 599270400, 599270400, 599356800, 599356800, 599443200, 599443200,
+							599529600, 599529600, 599616000), .Dim = c(2L, 5L), cal = "365",
+					months = c(31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31), class = "PCICt", dpy = 365, tzone = "GMT", units = "secs"))
+			
+	lapply(f, function(f1) {
+				expect_equal(nc.get.time.series(f1), correct.data.ts.test1)
+				expect_equal(nc.get.time.series(f1, return.bounds=TRUE), correct.data.ts.test2)
+				expect_equal(nc.get.time.series(f1, "tasmax"), correct.data.ts.test1)
+				expect_equal(nc.get.time.series(f1, time.dim.name="time"), correct.data.ts.test1)
+						
+				expect_error(nc.get.time.series(f1, "foo"))
+				expect_error(nc.get.time.series(f1, time.dim.name="foo"))
+				nc_close(f1)
+			})		
+})
+
+test_that("grid mappings are read and applied", {
+	f <- nc_open("test1.nc", readunlim=FALSE)
+	proj4.string <- "+proj=ob_tran +o_proj=longlat +lon_0=-97 +o_lat_p=42.5 +a=1 +to_meter=0.0174532925199 +no_defs"
+	expect_equal(nc.get.proj4.string(f, "tasmax"), proj4.string)
+	
+	lat.dat <- ncvar_get(f, "lat")
+	expect_equal(dim(lat.dat), c(155,130))
+	expect_equal(lat.dat[1,1], 12.3573083877563)
+	expect_equal(lat.dat[1,130], 59.1459007263184)
+	expect_equal(lat.dat[155,1], 12.3573160171509)
+	expect_equal(lat.dat[155,130], 59.1459159851074)
+	
+	
+	lon.dat <- ncvar_get(f, "lon")
+	expect_equal(dim(lon.dat), c(155,130))
+	expect_equal(lon.dat[1,1], 232.930877685547)
+	expect_equal(lon.dat[1,130], 189.603073120117)
+	expect_equal(lon.dat[155,1], 293.069122314453)
+	expect_equal(lon.dat[155,130], 336.396881103516)
+			
+	indices <- matrix(c(1, 155, 155, 1, 1, 1, 130, 130), nrow=4, ncol=2)
+	colnames(indices) <- c("x", "y")
+	
+	# The project() function in the proj4 r package returns unexpected values
+	# on these inputs on Debian systems.
+	# Run on Ubuntu, Windows, Fedora, CentOS, and Mac, the tests behave as expected.
+	# Accordingly, the last part of this test is skipped on Linux systems.
+	skip_on_os("linux")
```

Comment:
Everything here looks good to me. My only question is regarding "linux" as the `skip_on_os()` parameter. Ubuntu, Fedora, and CentOS *are* all Linux systems. Does `skip_on_os()` have more specific labels for them or something? `"linux" == Debian` as far as this function goes?

Reply:
Unfortunately, the [possible values](https://rdrr.io/cran/testthat/man/skip.html) for `skip_on_os()` are "windows", "mac", "linux" and "solaris". So tests on other linux systems are skipped as well, even though they'd pass if we did them.

Reply:
From the shell `$ lsb_release -si` should tell you.

Reply:
`Sys.info()` inside R seems promising. 
```R
> Sys.info()
                                       sysname 
                                       "Linux" 
                                       release 
                          "4.15.0-101-generic" 
                                       version 
"#102-Ubuntu SMP Mon May 11 10:07:26 UTC 2020" 
                                      nodename 
                                   "pcic-2023" 
                                       machine 
                                      "x86_64" 
                                         login 
                                      "lzeman" 
                                          user 
                                      "lzeman" 
                                effective_user 
                                      "lzeman" 
```